Mathlib/RingTheory/MvPowerSeries/Order.lean

```diff
@@ -0,0 +1,525 @@
+theorem weightedOrder_monomial (d : σ →₀ ℕ) (a : R) [Decidable (a = 0)] :
+    weightedOrder w (monomial R d a) = if a = 0 then (⊤ : ℕ∞) else weight w d := by
+  classical
+  split_ifs with h
+  · rw [h, weightedOrder_eq_top_iff, LinearMap.map_zero]
+  · rw [weightedOrder_eq_nat]
+    constructor
+    · use d; simp only [coeff_monomial_same, eq_self_iff_true, ne_eq, true_and_iff]; exact h
+    · intro b hb
+      rw [coeff_monomial, if_neg]
+      intro h
+      simp only [h, lt_self_iff_false] at hb
+
+/-- The order of the monomial `a*X^n` is `n` if `a ≠ 0`.-/
+theorem weightedOrder_monomial_of_ne_zero (d : σ →₀ ℕ) (a : R) (h : a ≠ 0) :
+    weightedOrder w (monomial R d a) = weight w d := by
+  classical
+  rw [weightedOrder_monomial, if_neg h]
+
+
+/-- The order of the sum of two formal power series is at least the minimum of their orders.-/
+theorem le_weightedOrder_add (f g : MvPowerSeries σ R) :
+    min (f.weightedOrder w) (g.weightedOrder w) ≤ (f + g).weightedOrder w := by
+  apply le_weightedOrder w
+  simp (config := { contextual := true }) only [coeff_of_lt_weightedOrder w, lt_min_iff, map_add,
+    add_zero, eq_self_iff_true, imp_true_iff]
+
+private theorem weightedOrder_add_of_weightedOrder_lt.aux {f g : MvPowerSeries σ R}
+    (H : f.weightedOrder w < g.weightedOrder w) :
+    (f + g).weightedOrder w = f.weightedOrder w := by
+  obtain ⟨n, hn⟩ := ne_top_iff_exists.mp (ne_top_of_lt H)
+  erw [← hn, weightedOrder_eq_nat]
+  obtain ⟨d, hd, hd'⟩ := ((weightedOrder_eq_nat w).mp hn.symm).1
+  constructor
+  · use d; use hd
+    rw [← hn, ← hd] at H
+    rw [(coeff _ _).map_add, coeff_of_lt_weightedOrder w g H, add_zero]
+    exact hd'
+  · intro b hb
+    suffices ↑(weight w b) < weightedOrder w f by
+      rw [(coeff _ _).map_add, coeff_of_lt_weightedOrder w f this,
+        coeff_of_lt_weightedOrder w g (lt_trans this H), add_zero]
+    rw [← hn, ENat.some_eq_coe, Nat.cast_lt]
+    exact hb
+
+/-- The weighted_order of the sum of two formal power series
+ is the minimum of their orders if their orders differ.-/
+theorem weightedOrder_add_of_weightedOrder_ne {f g : MvPowerSeries σ R}
+    (h : f.weightedOrder w ≠ g.weightedOrder w) :
+    weightedOrder w (f + g) = weightedOrder w f ⊓ weightedOrder w g := by
+  refine le_antisymm ?_ (le_weightedOrder_add w _ _)
+  by_cases H₁ : f.weightedOrder w < g.weightedOrder w
+  · simp only [le_inf_iff, weightedOrder_add_of_weightedOrder_lt.aux w H₁]
+    exact ⟨le_rfl, le_of_lt H₁⟩
+  · by_cases H₂ : g.weightedOrder w < f.weightedOrder w
+    · simp only [add_comm f g, le_inf_iff, weightedOrder_add_of_weightedOrder_lt.aux w H₂]
+      exact ⟨le_of_lt H₂, le_rfl⟩
+    · exact absurd (le_antisymm (not_lt.1 H₂) (not_lt.1 H₁)) h
+
+/-- The weighted_order of the product of two formal power series
+ is at least the sum of their orders.-/
+theorem weightedOrder_mul_ge [DecidableEq σ] (f g : MvPowerSeries σ R) :
```

Comment:
As with the comment on the other PR: please state the name with `le`
```suggestion
theorem add_weightedOrder_le_weightedOrder_mul [DecidableEq σ] (f g : MvPowerSeries σ R) :
```
or
```suggestion
theorem le_weightedOrder_mul [DecidableEq σ] (f g : MvPowerSeries σ R) :
```
The current name could be an alias.

Reply:
I chose the second one.

Reply:
Also : I added the initial name as an alias, and did a similar modification for the already existing `PowerSeries.order_mul_ge`